lib/services/ui_config.go

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// UnmarshalUIConfig unmarshals the UIConfig resource from JSON.
+func UnmarshalUIConfig(data []byte, opts ...MarshalOption) (types.UIConfig, error) {
```

Comment:
as the marshaller has a check for `v1` do you think the unmarshaller should also be checking for `v1`?